packages/wallet-frontend/src/pages/transfer/pay.tsx

```diff
@@ -13,26 +13,70 @@ import { accountService } from '@/lib/api/account'
 import { paySchema, transfersService } from '@/lib/api/transfers'
 import { useDialog } from '@/lib/hooks/useDialog'
 import { SuccessDialog } from '@/components/dialogs/SuccessDialog'
-import { getObjectKeys } from '@/utils/helpers'
-import { useState } from 'react'
+import { getObjectKeys, transformAmount } from '@/utils/helpers'
+import { useEffect, useState } from 'react'
 import { paymentPointerService } from '@/lib/api/paymentPointer'
 import { ErrorDialog } from '@/components/dialogs/ErrorDialog'
+import { useDebounce } from '@/lib/hooks/useDebounce'
+import { assetService } from '@/lib/api/asset'
 
 type PayProps = InferGetServerSidePropsType<typeof getServerSideProps>
 
 export default function Pay({ accounts }: PayProps) {
+  let scale = 2
   const [openDialog, closeDialog] = useDialog()
   const [paymentPointers, setPaymentPointers] = useState<SelectOption[]>([])
   const [balance, setBalance] = useState('')
+  const [incomingPaymentUrl, setIncomingPaymentUrl] = useState('')
+  const [debouncedIncomingPaymentUrl, isLoading] = useDebounce(
+    incomingPaymentUrl,
+    1000
+  )
+
   const payForm = useZodForm({
     schema: paySchema
   })
 
+  const fetchData = async () => {
+    const tempArray = incomingPaymentUrl.split('/')
+    const incomingPaymentId = tempArray[tempArray.length - 1]
+    const response = await transfersService.getIncomingPaymentDetails(
+      incomingPaymentId
+    )
+    if (response.success && response.data) {
+      const { value, description } = response.data
+      payForm.setValue('amount', transformAmount(value, scale))
+      payForm.setValue('description', description ?? '')
+    } else {
+      const { message } = response
+      payForm.setError('root', { message })
+      payForm.setValue('amount', 0)
+      payForm.setValue('description', '')
+    }
+  }
+
+  useEffect(() => {
+    if (debouncedIncomingPaymentUrl) {
+      fetchData()
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [debouncedIncomingPaymentUrl])
```

Comment:
How about using a `DebouncedInput` component instead? That way, we might be able to eliminate this `useEffect`. 🤔